clientv3/balancer.go

```diff
@@ -354,3 +363,47 @@ func getHost(ep string) string {
 	}
 	return url.Host
 }
+
+func (b *simpleBalancer) switchEndpoint() {
+	tNow := time.Now()
+	b.mu.RLock()
+
+	addrs := make([]grpc.Address, 0, len(b.addrs))
+	lastSwitchTime := b.lastSwitchTime
+
+	b.mu.RUnlock()
+
+	if lastSwitchTime.Add(30 * time.Second).After(tNow) {
```

Comment:
graylist has one problem, what time should we put the graylisted endpoint back to noramal? in the client we don't have enough info

I think a good way it to put a counter on each endpoint, every time switch the endpoint we choose a minimal retry count endpoint to connect. And all counter will reset every time updateEndpoints happened, periodically.

Still this will change a lot of code in balancer now. and it may cause some kind of un-balance between multiple clients. after a period maybe most client will attach to a health member and leave sick-backto-health member without clients.

The algorithm now is random choose. when a client found a good endpoint, it will stick to it. the random choose is near to balance. as once it found a health member it will stick to it. and will be low probability cause 'bouncing between both partitioned endpoints'